packages/flutter/lib/src/material/time_picker_theme.dart

```diff
@@ -0,0 +1,389 @@
+        && other.inputDecorationTheme == inputDecorationTheme;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(ColorProperty('backgroundColor', backgroundColor, defaultValue: null));
+    properties.add(ColorProperty('hourMinuteSelectedTextColor', hourMinuteSelectedTextColor, defaultValue: null));
+    properties.add(ColorProperty('hourMinuteSelectedColor', hourMinuteSelectedColor, defaultValue: null));
+    properties.add(ColorProperty('hourMinuteUnselectedTextColor', hourMinuteUnselectedTextColor, defaultValue: null));
+    properties.add(ColorProperty('hourMinuteUnselectedColor', hourMinuteUnselectedColor, defaultValue: null));
+    properties.add(ColorProperty('dayPeriodSelectedTextColor', dayPeriodSelectedTextColor, defaultValue: null));
+    properties.add(ColorProperty('dayPeriodSelectedColor', dayPeriodSelectedColor, defaultValue: null));
+    properties.add(ColorProperty('dayPeriodUnselectedTextColor', dayPeriodUnselectedTextColor, defaultValue: null));
+    properties.add(ColorProperty('dayPeriodUnselectedColor', dayPeriodUnselectedColor, defaultValue: null));
+    properties.add(ColorProperty('dialHandColor', dialHandColor, defaultValue: null));
+    properties.add(ColorProperty('dialBackgroundColor', dialBackgroundColor, defaultValue: null));
+    properties.add(ColorProperty('dayPeriodBorderColor', dayPeriodBorderColor, defaultValue: null));
+    properties.add(DiagnosticsProperty<TextStyle>('hourMinuteTextStyle', hourMinuteTextStyle, defaultValue: null));
+    properties.add(DiagnosticsProperty<TextStyle>('dayPeriodTextStyle', dayPeriodTextStyle, defaultValue: null));
+    properties.add(DiagnosticsProperty<TextStyle>('helpTextStyle', helpTextStyle, defaultValue: null));
+    properties.add(DiagnosticsProperty<ShapeBorder>('shape', shape, defaultValue: null));
+    properties.add(DiagnosticsProperty<ShapeBorder>('hourMinuteShape', hourMinuteShape, defaultValue: null));
+    properties.add(DiagnosticsProperty<ShapeBorder>('dayPeriodShape', dayPeriodShape, defaultValue: null));
+    properties.add(DiagnosticsProperty<InputDecorationTheme>('inputDecorationTheme', inputDecorationTheme, defaultValue: null));
+  }
+}
+
+/// An inherited widget that defines the configuration for time pickers
+/// displayed in this widget's subtree.
+///
+/// Values specified here are used for time picker properties that are not
+/// given an explicit non-null value.
+class TimePickerTheme extends InheritedTheme {
+  /// Creates a time picker theme that controls the configurations for
+  /// time pickers displayed in its widget subtree.
+  const TimePickerTheme({
+    Key key,
+    this.data,
+    Widget child,
+  }) : super(key: key, child: child);
+
+  /// The properties for descendant time picker widgets.
+  final TimePickerThemeData data;
+
+  /// The closest instance of this class's [data] value that encloses the given
```

Comment:
If you read this carefully, it's not really correct.

Reply:
I'm not sure what you mean, could you elaborate?

Reply:
I think you mean:

The [data] value of the closest instance of this class that encloses this context.

Or you could just say

The [data] value of the closes [TimePickerTheme] ancestor.

Reply:
Ohhhhh that makes more sense, fixed!